Source/WebCore/rendering/style/StyleGradientImage.h

```diff
@@ -103,15 +102,12 @@ class StyleGradientImage final : public StyleGeneratedImage {
 
     template<typename GradientAdapter> GradientColorStops computeStops(GradientAdapter&, const CSSToLengthConversionData&, const RenderStyle&, float maxLengthForRepeat) const;
 
-    bool hasColorDerivedFromElement() const;
-    bool isCacheable() const;
-
     Data m_data;
     CSSGradientRepeat m_repeat;
     CSSGradientType m_gradientType;
     CSSGradientColorInterpolationMethod m_colorInterpolationMethod;
-    Vector<Stop> m_stops;
-    mutable std::optional<bool> m_hasColorDerivedFromElement;
+    Vector<StyleGradientImageStop> m_stops;
+    bool m_knownCacheableBarringFilter;
```

Comment:
Would be nice to initialize this here even if that’s never relied on in practice. I would suggest we do it with all scalars unless we are trying to optimize something.

Reply:
Seems reasonable. I kinda wish we (and maybe this exists) we could just enforce that all fields must be initialized at the end of construction (with some ability to disable when needed), but maybe c++ initialization patterns are too complicated / cross translation unity to allow for that kind of analysis (Swift has made me soft).